File: semester 4/MPP/JFTS/FestivalPersistance/src/main/java/ro/mpp/repository/DBRepository/VenueRepository.java
package ro.mpp.repository.DBRepository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp.domain.Venue;
import ro.mpp.repository.IVenueRepository;
import ro.mpp.utils.Database;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class VenueRepository implements IVenueRepository {
    private final Database db = new Database();
    private static final Logger logger = LogManager.getLogger(VenueRepository.class);

    @Override
    public Optional<Venue> save(Venue entity) {
        logger.traceEntry("VenueRepository save {}", entity);
        String sql = "INSERT INTO venue (name, address, capacity) VALUES (?, ?, ?)";

        try {
            Connection conn = db.getConnection();
            try (PreparedStatement ps = conn.prepareStatement(sql, PreparedStatement.RETURN_GENERATED_KEYS)) {
                ps.setString(1, entity.getName());
                ps.setString(2, entity.getAddress());
                ps.setInt(3, entity.getCapacity());

                ps.executeUpdate();

                try (ResultSet rs = ps.getGeneratedKeys()) {
                    if (rs.next()) {
                        entity.setId(rs.getInt(1));
                    }
                }
                return Optional.of(entity);
            }
        }catch (SQLException e) {
            logger.error("Error while saving venue", e);
            throw new RuntimeException("Error while saving venue: " + e.getMessage());
        }
    }

    @Override
    public Optional<Venue> update(Venue entity) {
        logger.traceEntry("VenueRepository update {}", entity);
        String sql = "UPDATE venue SET name = ?, address = ?, capacity = ? WHERE id = ?";

        try {
            Connection conn = db.getConnection();
            try (PreparedStatement ps = conn.prepareStatement(sql)) {
                ps.setString(1, entity.getName());
                ps.setString(2, entity.getAddress());
                ps.setInt(3, entity.getCapacity());
                ps.setInt(4, entity.getId());

                ps.executeUpdate();
                return Optional.of(entity);
            }
        } catch (SQLException e) {
            logger.error("Error while updating venue", e);
            throw new RuntimeException("Error while updating venue: " + e.getMessage());
        }
    }

    @Override
    public void delete(Venue entity) {
        logger.traceEntry("VenueRepository delete {}", entity);
        String sql = "DELETE FROM venue WHERE id = ?";

        try {
            Connection conn = db.getConnection();
            try (PreparedStatement ps = conn.prepareStatement(sql)) {

                ps.setInt(1, entity.getId());
                ps.executeUpdate();
            }
        } catch (SQLException e) {
            logger.error("Error while deleting venue", e);
            throw new RuntimeException("Error while deleting venue: " + e.getMessage());
        }
    }

    @Override
    public Optional<Venue> find(Integer id) {
        logger.traceEntry("VenueRepository find venue with id {}", id);
        String sql = "SELECT * FROM venue WHERE id = ?";

        try {
            Connection conn = db.getConnection();
            try (PreparedStatement ps = conn.prepareStatement(sql)) {

                ps.setInt(1, id);
                try (ResultSet rs = ps.executeQuery()) {
                    if (rs.next()) {
                        return Optional.of(new Venue(rs));
                    }
                }
            }
        } catch (SQLException e) {
            logger.error("Error while finding venue with id {}", id, e);
            throw new RuntimeException("Error while finding venue: " + e.getMessage());
        }
        return Optional.empty();
    }

    @Override
    public List<Venue> findAll() {
        logger.traceEntry("VenueRepository findAll");
        String sql = "SELECT * FROM venue";
        List<Venue> venues = new ArrayList<>();

        try {
            Connection conn = db.getConnection();
            try (PreparedStatement ps = conn.prepareStatement(sql)) {

                try (ResultSet rs = ps.executeQuery()) {
                    while (rs.next()) {
                        venues.add(new Venue(rs));
                    }
                }
                return venues;
            }
        } catch (SQLException e) {
            logger.error("Error while finding all venues", e);
            throw new RuntimeException("Error while finding all venues: " + e.getMessage());
        }
    }
}

File: semester 4/MPP/C#FTS/Model/Repository/DBRepository/VenueRepository.cs
using C_FTS.Domain;
using C_FTS.Utils;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Text;

namespace C_FTS.Repository.DBRepository
{
    internal class VenueRepository : IVenueRepository
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(VenueRepository));
        private Database db = new Database();
        public void Delete(Venue entity)
        {
            throw new NotImplementedException();
        }

        public Venue? Find(int id)
        {
            logger.DebugFormat("Entering Find with id {0}", id);
            IDbConnection conn = db.GetConnection();

            using (IDbCommand cmd = conn.CreateCommand())
            {
                cmd.CommandText = "SELECT * FROM venue WHERE id = @id";
                IDbDataParameter dp = cmd.CreateParameter();
                dp.ParameterName = "@id";
                dp.Value = id;
                cmd.Parameters.Add(dp);

                using (var dataR = cmd.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        Venue venue = new Venue(dataR);
                        logger.DebugFormat("Exiting Find with {0}", venue);
                        return venue;
                    }
                    logger.Error("Exiting Find with {0}", null);
                    return null;
                }
            }
        }

        public List<Venue> FindAll()
        {
            throw new NotImplementedException();
        }

        public Venue? Save(Venue entity)
        {
            throw new NotImplementedException();
        }

        public Venue? Update(Venue entity)
        {
            throw new NotImplementedException();
        }
    }
}
